### PR TITLE
[String] Documented the s() function

### LINKED ISSUE #13068: [String] Add the s() helper method
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#35625
| PR author(s) | @fancyweb
| Merged in    | 5.1-dev

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #13068.

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

**Review by HeahDude — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Javier.

File: components/string.rst [hunk 1]
```diff
@@ -104,20 +104,35 @@ Use the ``wrap()`` static method to instantiate more than one string object::
...
+    // both lines are equivalent
+    $foo = new UnicodeString('hello');
+    $foo = u('hello');
+
+    // the s() function creates a byte string or Unicode string
+    // depending on the given contents
+    use function Symfony\Component\String\u;
+
+    // creates a ByteString object
+    $foo = s('\xfe\xff');
```
Comment (nicolas-grekas) [inline components/string.rst]: Double quotes would be needed for escaped chars to be parsed

Comment (OskarStark) [reply components/string.rst]: Will fix it

Comment (OskarStark) [reply components/string.rst]: https://github.com/symfony/symfony-docs/pull/13122

**Pull request merged**